src/Interpreters/GinFilter.h

```diff
@@ -54,15 +32,88 @@ struct GinSegmentWithRowIdRange
 
 using GinSegmentWithRowIdRangeVector = std::vector<GinSegmentWithRowIdRange>;
 
+
 /// GinFilter provides underlying functionalities for building text index and also
 /// it does filtering the unmatched rows according to its query string.
 /// It also builds and uses skipping index which stores (segment_id, rowid_start, rowid_end) triples.
 class GinFilter
 {
 public:
-    GinFilter() = default;
 
-    GinFilter(std::string_view query_string_, const std::vector<String> & search_terms_);
+    struct Parameters
+    {
+        Parameters(
+            String tokenizer_,
+            UInt64 segment_digestion_threshold_bytes_,
+            double bloom_filter_false_positive_rate_,
+            std::optional<UInt64> ngram_size_,
+            std::optional<std::vector<String>> separators_);
+
+        String tokenizer;
+        /// Digestion threshold to split a segment. By default, it is 0 (zero) which means unlimited.
+        UInt64 segment_digestion_threshold_bytes;
+        /// Bloom filter false positive rate, by default it's 0.1%.
+        double bloom_filter_false_positive_rate;
+        /// For ngram tokenizer
+        std::optional<UInt64> ngram_size;
+        /// For split tokenizer
+        std::optional<std::vector<String>> separators;
+
+        bool operator<=>(const Parameters& other) const = default;
+    };
+
+    class StringInfo
```

Comment:
"StringInfo" sounds quite generic. Maybe "QueryString" (which is what the class contains, as well as terms which are derived from the query string)? Up to you.

Reply:
I was concerned about readability, I know they won't conflict (AWS was just an example)